processor/schemaprocessor/processor.go

```diff
@@ -89,7 +89,45 @@ func (t schemaProcessor) processLogs(ctx context.Context, ld plog.Logs) (plog.Lo
 	return ld, nil
 }
 
-func (t schemaProcessor) processMetrics(_ context.Context, md pmetric.Metrics) (pmetric.Metrics, error) {
+func (t schemaProcessor) processMetrics(ctx context.Context, md pmetric.Metrics) (pmetric.Metrics, error) {
+	for mt := 0; mt < md.ResourceMetrics().Len(); mt++ {
+		rMetric := md.ResourceMetrics().At(mt)
+		resourceSchemaURL := rMetric.SchemaUrl()
+		if resourceSchemaURL != "" {
+			t.log.Debug("requesting translation for resourceSchemaURL", zap.String("resourceSchemaURL", resourceSchemaURL))
+			tr, err := t.manager.RequestTranslation(context.Background(), resourceSchemaURL)
+			if err != nil {
+				t.log.Error("failed to request translation", zap.Error(err))
+				return md, err
+			}
+			err = tr.ApplyAllResourceChanges(rMetric, resourceSchemaURL)
+			if err != nil {
+				t.log.Error("failed to apply resource changes", zap.Error(err))
+				return md, err
+			}
+		}
+		for sm := 0; sm < rMetric.ScopeMetrics().Len(); sm++ {
+			metric := rMetric.ScopeMetrics().At(sm)
+			metricSchemaURL := metric.SchemaUrl()
+			if metricSchemaURL == "" {
+				metricSchemaURL = resourceSchemaURL
+			}
+			if metricSchemaURL == "" {
+				continue
+			}
+			tr, err := t.manager.
+				RequestTranslation(ctx, metricSchemaURL)
+			if err != nil {
+				t.log.Error("failed to request translation", zap.Error(err))
+				continue
+			}
+			err = tr.ApplyScopeMetricChanges(metric, metricSchemaURL)
+			if err != nil {
+				t.log.Error("failed to apply scope metric changes", zap.Error(err))
```

Comment:
you don't want to return errors?

Reply:
Thanks for pointing this out. I have updated it to return errors. I have added a test case for the same as well. I need to make a similar change for other signals. I will do it in a follow-up PR.